make feeOpt dependent on minimum utxo

paweljakubas opened the pull request:
# Issue Number

<!-- Put here a reference to the issue this PR relates to and which requirements it tackles -->


# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- [x] I have make feeOpts dependent on minUtxoval
- [ ] I have made fee balancing algorithm be aware of dust threshold

# Comments

<!-- Additional comments or screenshots to attach if any -->

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Assign the PR to a corresponding milestone
 ✓ Acknowledge any changes required to the Wiki
-->

paweljakubas committed:
make feeOpt dependent on minimum utxo

KtorZ committed:
add non-null dust & property reflecting it for 'rebalanceSelection'

  This fails now, since 'rebalanceSelection' completely disregard any
  form of dust threshold.

KtorZ committed:
acknowledge dust threshold in fee balancing

  I greatly simplified the function as well, it probably handles slightly worse some edge-cases but it's probably for the best in terms of maintenance burden.

KtorZ reviewed (approved):
Added refactor of the fee balancing function on top to get the dustThreshold acknowledged there too.

KtorZ committed:
revert refactor on 'rebalanceSelection' and coalesce dust AFTER rebalancing

  What we loose from this might be exact balancing in some cases (when the transaction is exactly balanced but coalescing dust may remove one or more change output and change the fee).
  So we might hit problems with some edge-cases on Jörmungandr but things should be fine on cardano-node where we might occasionally pay a little bit more than necessary...
